Index: llvm-toolchain-snapshot_21~++20250211013808+e258bca9505f/libc/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_21~++20250211013808+e258bca9505f.orig/libc/CMakeLists.txt
+++ llvm-toolchain-snapshot_21~++20250211013808+e258bca9505f/libc/CMakeLists.txt
@@ -51,7 +51,11 @@ set(LIBC_KERNEL_HEADERS "/usr/include" C
 # Defining a global namespace to enclose all libc functions.
 set(default_namespace "__llvm_libc")
 if(LLVM_VERSION_MAJOR)
-  string(REPLACE "-" "" NS_LLVM_VERSION_SUFFIX ${LLVM_VERSION_SUFFIX})
+  if(DEFINED LLVM_VERSION_SUFFIX AND NOT "${LLVM_VERSION_SUFFIX}" STREQUAL "")
+    string(REPLACE "-" "" NS_LLVM_VERSION_SUFFIX "${LLVM_VERSION_SUFFIX}")
+  else()
+    set(NS_LLVM_VERSION_SUFFIX "")
+  endif()
   set(default_namespace "__llvm_libc_${LLVM_VERSION_MAJOR}_${LLVM_VERSION_MINOR}_${LLVM_VERSION_PATCH}_${NS_LLVM_VERSION_SUFFIX}")
 endif()
 set(LIBC_NAMESPACE ${default_namespace}
