Index: llvm-toolchain-snapshot_21~++20250630094904+57f7e14b5779/llvm/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_21~++20250630094904+57f7e14b5779.orig/llvm/CMakeLists.txt
+++ llvm-toolchain-snapshot_21~++20250630094904+57f7e14b5779/llvm/CMakeLists.txt
@@ -1019,7 +1019,7 @@ set(LLVM_PROFDATA_FILE "" CACHE FILEPATH
 
 if(LLVM_INCLUDE_TESTS)
   # All LLVM Python files should be compatible down to this minimum version.
-  set(LLVM_MINIMUM_PYTHON_VERSION 3.8)
+  set(LLVM_MINIMUM_PYTHON_VERSION 3.5)
 else()
   # FIXME: it is unknown if this is the actual minimum bound
   set(LLVM_MINIMUM_PYTHON_VERSION 3.0)
